flink-yarn-tests/src/test/java/org/apache/flink/yarn/testjob/YarnTestArchiveJob.java

```diff
@@ -0,0 +1,137 @@
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * Testing job for localizing resources of LocalResourceType.ARCHIVE in per job cluster mode.
+ */
+public class YarnTestArchiveJob {
+	private static final List<String> LIST = ImmutableList.of("test1", "test2");
+
+	private static final Map<String, String> srcFiles = new HashMap<String, String>() {{
+			put("local1.txt", "Local text Content1");
+			put("local2.txt", "Local text Content2");
+		}};
+
+	private static void archiveFilesInDirectory(File directory, String target) throws IOException {
+
+		for (Map.Entry<String, String> entry : srcFiles.entrySet()) {
+			Files.write(Paths.get(directory.getAbsolutePath() + File.separator + entry.getKey()),
+				entry.getValue().getBytes());
+		}
+
+		try (FileOutputStream fos = new FileOutputStream(target);
+				GzipCompressorOutputStream gos = new GzipCompressorOutputStream(new BufferedOutputStream(fos));
+				TarArchiveOutputStream taros = new TarArchiveOutputStream(gos)) {
+			taros.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU);
+			for (File f : directory.listFiles()) {
+				taros.putArchiveEntry(new TarArchiveEntry(f,
+					directory.getName() + File.separator + f.getName()));
+				try (FileInputStream fis = new FileInputStream(f);
+						BufferedInputStream bis = new BufferedInputStream(fis)) {
+					IOUtils.copy(bis, taros);
+					taros.closeArchiveEntry();
+				}
+			}
+		}
+	}
+
+	public static JobGraph getArchiveJobGraph(File testDirectory, Configuration config) throws IOException {
+
+		final String archive = testDirectory.getAbsolutePath().concat(".tar.gz");
+		final String localizedPath = testDirectory.getName().concat(".tar.gz") + File.separator + testDirectory.getName();
+
+		archiveFilesInDirectory(testDirectory, archive);
+
+		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+
+		env.addSource(new SourceFunctionWithArchive(LIST, localizedPath, TypeInformation.of(String.class)))
+			.setParallelism(1)
+			.addSink(new DiscardingSink<String>())
+			.setParallelism(1);
+		config.set(YarnConfigOptions.SHIP_ARCHIVES, Collections.singletonList(archive));
+
+		return env.getStreamGraph().getJobGraph();
+	}
+
+	private static class SourceFunctionWithArchive<T> extends RichSourceFunction<T> implements ResultTypeQueryable<T> {
+		private List<T> inputDataset;
+		private String resourcePath;
+		private TypeInformation returnType;
+
+		public void open(Configuration parameters) throws Exception {
+			for (Map.Entry<String, String> entry : srcFiles.entrySet()) {
+				String content = new String(Files.readAllBytes(Paths.get(resourcePath + File.separator + entry.getKey())));
+				checkArgument(entry.getValue().equals(content), "The content of the unpacked file should be identical to the original file's.");
```

Comment:
Even if I put here `Paths.get(resourcePath + File.separator + entry.getKey() + "adssad"` the test will still pass but the logs say that it could not find the file (as expected).